tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py

```diff
@@ -0,0 +1,123 @@
+            os.path.dirname(__file__),
+            'pvs_studio_output_test_files'
+        )
+        self.result_dir = tempfile.mkdtemp()
+
+    def tearDown(self) -> None:
+        """Clean temporary directory. """
+        shutil.rmtree(self.result_dir)
+
+    def test_no_report_output_file(self) -> None:
+        """ Test transforming single cpp file. """
+        result = os.path.join(self.test_files, "files", "sample.cpp")
+
+        is_success = self.analyzer_result.transform(
+            analyzer_result_file_paths=[result],
+            output_dir_path=self.result_dir,
+            export_type=plist.EXTENSION,
+            file_name="{source_file}_{analyzer}"
+        )
+
+        self.assertFalse(is_success)
+
+    def test_transform_dir(self) -> None:
+        """ Test transforming a directory. """
+        result = os.path.join(self.test_files)
+
+        is_success = self.analyzer_result.transform(
+            analyzer_result_file_paths=[result],
+            output_dir_path=self.result_dir,
+            export_type=plist.EXTENSION,
+            file_name="{source_file}_{analyzer}"
+        )
+
+        self.assertFalse(is_success)
+
+    def test_transform_single_file(self) -> None:
+        """ Test transforming single output file. """
+        self.make_report_valid()
+        result = os.path.join(self.test_files, 'sample.json')
+
+        is_success = self.analyzer_result.transform(
+            analyzer_result_file_paths=[result],
+            output_dir_path=self.result_dir,
+            export_type=plist.EXTENSION,
+            file_name="{source_file}_{analyzer}"
+        )
+
+        self.assertTrue(is_success)
+
+        plist_file = os.path.join(self.result_dir,
+                                  'sample.cpp_pvs-studio.plist')
+
+        with open(plist_file, mode='rb') as pfile:
+            res = plistlib.load(pfile)
+            res['files'][0] = os.path.join('files', 'sample.cpp')
+
+        plist_file = os.path.join(self.test_files,
+                                  'sample.plist')
+        with open(plist_file, mode='rb') as pfile:
+            exp = plistlib.load(pfile)
+
+        self.assertEqual(
+            res["diagnostics"][0]["check_name"],
+            exp["diagnostics"][0]["check_name"]
+        )
+
+        self.assertEqual(
+            res["diagnostics"][0]["location"]["line"],
+            exp["diagnostics"][0]["location"]["line"]
+        )
+
+    @staticmethod
+    def make_report_valid() -> None:
+        samples_path = os.path.join(
+            os.path.dirname(__file__),
+            "pvs_studio_output_test_files"
+        )
+
+        path_to_file = os.path.join(
+            samples_path,
+            "files",
+            "sample.cpp"
+        )
+
+        report_path = os.path.join(samples_path, "sample.json")
+        with open(report_path, 'r') as file:
+            data = json.loads(file.read())
+            data["warnings"][0]["positions"][0]["file"] = path_to_file
```

Comment:
does PVS studio support relative path to the source files?
If it does, I think we could test the transform() function for .json files with relative paths.

Reply:
No, PVS-Studio uses absolute paths to source files.